.github/workflows/rust.yml

```diff
@@ -0,0 +1,42 @@
+name: Rust
+
+on:
+  push:
+  pull_request:
+    branches: [ "main" ]
```

Comment:
There are no branches specified in `push` section - is it going to run CI on each push to every branch?
I think we can restrict it to `main` branch, same as with `pull_request` - that's what other Scylla projects do as far as I can see:
https://github.com/scylladb/cpp-rust-driver/blob/9999af005726b9d47d1837536e58d463ec63da63/.github/workflows/build.yml#L4-L7
https://github.com/scylladb/java-driver/blob/3e2d8a1766150d78bd806264ecf1e1870e0f14cf/.github/workflows/tests%40v1.yml#L4-L7
https://github.com/scylladb/scylla-rust-driver/blob/9609a4895a393818e68565d236ebe7460fea6e1c/.github/workflows/cassandra.yml#L5-L8